### PR TITLE
Several small idiomatic changes to the edn crate

### PR DESCRIPTION
**Pull request by victorporof**

### CODE REVIEW

File: edn/src/types.rs [hunk 1]
```diff
@@ -199,15 +199,13 @@ fn to_ord(value: &Value) -> i32 {
...
 }
 
 pub fn to_keyword(namespace: Option<&str>, name: &str) -> Value {
-    if let Some(ns) = namespace {
-        return Value::NamespacedKeyword(symbols::NamespacedKeyword::new(ns, name));
-    }
-    return Value::Keyword(symbols::Keyword::new(name));
+    namespace.map_or_else(
+        || Value::Keyword(symbols::Keyword::new(name)),
+        |ns| Value::NamespacedKeyword(symbols::NamespacedKeyword::new(ns, name)))
```
Comment (rnewman) [inline edn/src/types.rs]: This is a great spot to add a short doc test.

File: edn/tests/tests.rs [hunk 2]
```diff
@@ -946,8 +944,8 @@ fn test_is_and_as_type_helper_functions() {
             value.is_map(),
         ];
 
-        for j in 0..values.len() {
-            assert_eq!(j == i, is_result[j]);
+        for (j, result) in is_result.iter().enumerate() {
```
Comment (rnewman) [inline edn/tests/tests.rs]: If you're going to do this, you should also `assert_eq!(values.len(), is_result.len())` outside this inner loop. Previously the array indexing bounds check would panic if these arrays were mis-sized.

File: edn/src/types.rs (hunk 1 above)
Comment (victorporof) [inline edn/src/types.rs]: Will add.

File: edn/tests/tests.rs (hunk 2 above)
Comment (victorporof) [inline edn/tests/tests.rs]: Good idea.

**Comment by victorporof** [comment]
Addressed comments and rebased.

**Pull request merged**